src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParserStateObject.netcore.cs

```diff
@@ -1793,6 +1828,16 @@ private void SetBufferSecureStrings()
                             str = Marshal.SecureStringToBSTR(_securePasswords[i]);
                             byte[] data = new byte[_securePasswords[i].Length * 2];
                             Marshal.Copy(str, data, 0, _securePasswords[i].Length * 2);
+                            if (!BitConverter.IsLittleEndian)
+                            {
+                                byte temp;
+                                for (int ii = 0; ii < _securePasswords[i].Length * 2; ii += 2)
+                                {
+                                    temp = (byte)data[ii];
+                                    data[ii] = (byte)data[ii + 1];
+                                    data[ii + 1] = (byte)temp;
+                                }
+                            }
```

Comment:
An alternative way to do this would be:
```
for (int i = 0; i < data.Length; i += 2)
{
    var span = data.AsSpan(i);
    var value = BinaryPrimitives.ReadInt16LittleEndian(span);
    BinaryPrimitives.WriteInt16BigEndian(span, value);
}
```